functions/src/index.ts

```diff
@@ -103,11 +111,22 @@ app.get("/go/:id", (req, res) => {
             <meta property="og:url" content url="${fullUrl}"/>
             <meta property="og:title" content="${title}"/>
             <meta property="og:description" content="${description}"/>
-            <meta property="og:image" content="${image}" />
             <meta name="twitter:card" content="summary_large_image" />
             <meta property="twitter:title" content="${title}"/>
             <meta property="twitter:description" content="${description}"/>
-            <meta property="twitter:image" content="${image}"/>
+            ${
+              image &&
+              `<meta property="og:image" content="${image}" />
+              <meta property="twitter:image" content="${image}"/>`
+            }
+            ${
+              imgHeight &&
```

Comment:
adding image widths and heights makes it easier for the web crawlers to display the preview images (though I'm not 100% sure why)